packages/server-wallet/src/protocols/__test__/open-channel.test.ts

```diff
@@ -1,95 +1,210 @@
-  ${prefundState1}    | ${0}    | ${prefundState0}    | ${prefundState1}    | ${funded}    | ${signState(postfundState2)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
-  ${prefundState1}    | ${1}    | ${prefundState1}    | ${prefundState1}    | ${funded}    | ${signState(postfundState3)}   | ${'when the prefund state is supported, and the channel is funded'}      | ${'signs the postfund state'}
-  ${prefundState0}    | ${0}    | ${prefundState0}    | ${prefundState0}    | ${notFunded} | ${fundChannelAction1}          | ${'when the prefund state is supported, and alice is first to deposit'}  | ${'submits transaction'}
-  ${altPrefundState0} | ${0}    | ${altPrefundState0} | ${altPrefundState0} | ${funded}    | ${fundChannelAction2}          | ${'when the prefund state is supported, and alice is second to deposit'} | ${'submits transaction'}
-`('$result $cond', ({supported, latest, latestSignedByMe, funding, action, myIndex}) => {
-  const directFundingStatus = directFundingStatusFn(
-    supported,
-    funding,
-    participants[myIndex],
-    'Direct'
-  );
-  const ps = applicationProtocolState({
-    app: {
-      supported,
-      latest,
-      latestSignedByMe,
-      funding,
-      myIndex,
-      directFundingStatus,
-    },
+describe(`funding phase`, () => {
+  const statesHeld = [0, 1];
+
+  describe(`as participant 0`, () => {
+    it(`submits the transaction if it is my turn`, async () => {
+      const o = await testCase({participant: 0, statesHeld, totalFunds: 0, fundsToDeposit: 5});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it(`submits a top-up if there's a partial deposit`, async () => {
+      const o = await testCase({participant: 0, statesHeld, totalFunds: 2, fundsToDeposit: 3});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it(`does nothing if I've already deposited`, async () => {
+      await testCase({participant: 0, statesHeld, totalFunds: 5, fundsToDeposit: 0});
+    });
+  });
+  describe(`as participant 1`, () => {
+    it(`submits the transaction if it is my turn`, async () => {
+      const o = await testCase({participant: 1, statesHeld, totalFunds: 5, fundsToDeposit: 5});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it(`submits a top-up if there's a partial deposit`, async () => {
+      const o = await testCase({participant: 1, statesHeld, totalFunds: 7, fundsToDeposit: 3});
+      // and then doesn't submit it a second time
+      await testCrankOutcome(o, {fundsToDeposit: 0});
+    });
+    it.skip(`does nothing if I've already deposited`, async () => {
+      // todo: this fails, as it sends the post-fund-setup (as it probably should)
```

Comment:
I'm confused about whether this test should exist, be fixed, be skipped, or what?

Reply:
Me too. I'll remove it.